File: Classes/EnrollEase/CourseOffering.cpp
#include "CourseOffering.h"

namespace EnrollEase {

    CourseOffering::CourseOffering(int offeringID,
                                   const std::string& courseID,
                                   int semesterID,
                                   const std::string& instructor,
                                   const std::string& schedule)
      : offeringID(offeringID),
        courseID(courseID),
        semesterID(semesterID),
        instructor(instructor),
        schedule(schedule)
    {}

    int CourseOffering::getOfferingID() const {
        return offeringID;
    }

    std::string CourseOffering::getCourseID() const {
        return courseID;
    }

    int CourseOffering::getSemesterID() const {
        return semesterID;
    }

    std::string CourseOffering::getInstructor() const {
        return instructor;
    }

    std::string CourseOffering::getSchedule() const {
        return schedule;
    }

    void CourseOffering::setInstructor(const std::string& instr) {
        instructor = instr;
    }

    void CourseOffering::setSchedule(const std::string& sched) {
        schedule = sched;
    }

}

File: src/models/courseOffering.ts
export class CourseOffering {
  constructor(
    public offeringID: number,
    public courseID: string,
    public semesterID: number,
    public instructor: string,
    public schedule: string
  ) {}

  getOfferingID(): number {
    return this.offeringID;
  }

  getCourseID(): string {
    return this.courseID;
  }

  getSemesterID(): number {
    return this.semesterID;
  }

  getInstructor(): string {
    return this.instructor;
  }

  getSchedule(): string {
    return this.schedule;
  }

  setInstructor(instructor: string): void {
    this.instructor = instructor;
  }

  setSchedule(schedule: string): void {
    this.schedule = schedule;
  }
}
